### ISSUE TITLE
Idea: use tagged template literals for Catalog pages

### ISSUE THREAD

**Issue by herrstucki** [interactivethings/catalog#254]
Based on something @wereHamster mentioned a while back. The problem: when using Markdown documents, it's not possible to type-check/lint (with Flow) the Specimen code. This is possible with Catalog as React components (since it's "just" JS), although it makes writing text between the specimens much more tedious.

I think ES6 tagged template literals could be the sweet-spot here. Imagine writing Catalog docs like this:

```js
import React from 'react';
import {markdown, ReactSpecimen} from 'catalog';
import MyAwesomeComponent from './MyAwesomeComponent';

export default = () => markdown`

# Hello world

Foo bar [baz](/blah).

${
<ReactSpecimen>
  <MyAwesomeComponent />
</ReactSpecimen>
}

Blah blah blah etc.
`;
```

… or in a similar way. We could maybe even completely get rid of the tagged Markdown code block syntax.

Partly inspired by [markdown-in-js](https://github.com/threepointone/markdown-in-js) (although I wouldn't use that directly) and [next.js](https://github.com/zeit/next.js) (at least the page definition looks similar).

**Comment by naehrstoff** [comment]
I like it, but I'm not sure I understand what you mean by “get rid of the tagged Markdown code block syntax”.

If I remember correctly, the `markdown` helper would take an array of strings, and an array of values that are then interspersed between the strings. This would also be helpful for languages other than ES6.

**Comment by herrstucki** [comment]
What I meant was getting rid of parsing the plain markdown code blocks into specimens like we do it now. This of course would be a quite radical departure from the current way. Maybe it's not what we should do, but I kinda like the idea of looking at it with an open mind and thinking about what the _best_ way would be to use Catalog. And I think in its current form, it's either too low-level (when you use the Catalog components directly) or too abstract and indirect (e.g. when you write JSX code in a Markdown file and have to import stuff in the page configuration).

And yes, that's how tagged ES6 template literals work:

```js
foo`blah ${1} blubb ${"hey"}` // equivalent to  foo(["blah ", " blubb "], 1, "hey")
```

**Comment by naehrstoff** [comment]
Hickey would disapprove of getting rid of the existing ways, but I think the template literals would certainly help make it easier to mix components and docs.
 
<img width="1680" alt="avoid-breakage" src="https://cloud.githubusercontent.com/assets/10811/21063375/9d6d0c9c-be55-11e6-9121-dae917e30999.png">

**Comment by herrstucki** [comment]
Implemented in #277 and #281
